heat_infoblox/object_manipulator.py

```diff
@@ -87,6 +87,27 @@ def delete_member(self, member_name):
         member_data = {'host_name': member_name}
         self._delete_infoblox_object('member', member_data)
 
+    def add_member_dns_additional_ip(self, member_name, ip):
+        return_fields = ['additional_ip_list']
+        member_dns = self._get_member(member_name, return_fields,
+                                      fail_if_no_member=True,
+                                      object_type='member:dns')
+        additional_ips = member_dns.get('additional_ip_list') or []
+        additional_ips.append(ip)
+        payload = {'additional_ip_list': additional_ips}
+        self._update_infoblox_object_by_ref(member_dns['_ref'], payload)
+
+    def remove_member_dns_additional_ip(self, member_name, ip):
+        return_fields = ['additional_ip_list']
+        member_dns = self._get_member(member_name, return_fields,
+                                      object_type='member:dns')
+        if not member_dns or not member_dns.get('additional_ip_list'):
+            return
+        updated_ips = [orig_ip for orig_ip in member_dns['additional_ip_list']
```

Comment:
If there are multiple any cast resources I think you'll get a race condition here. You should probably use a lock.